### PR TITLE
Readme update with project scope

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriram@us.ibm.com>

### CODE REVIEW

**Comment by rshriram** [comment]
cc @mattklein123

**Comment by mattklein123** [comment]
LGTM. @envoyproxy/maintainers

**Comment by bndw** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,2 +1,33 @@
 # control-plane
-Go implementation of data-plane-api
+This repository contains a Go-based implementation of an API server that
+implements the discovery service APIs defined in
+[data-plane-api](https://github.com/envoyproxy/data-plane-api).
+
+## Scope
+
+However, due to the variety of platforms out there, there is no single
```
Comment (9len) [inline README.md]: Is there a sentence missing here? If not, cut "However"

File: README.md [hunk 2]
```diff
@@ -1,2 +1,33 @@
 # control-plane
-Go implementation of data-plane-api
+This repository contains a Go-based implementation of an API server that
+implements the discovery service APIs defined in
+[data-plane-api](https://github.com/envoyproxy/data-plane-api).
+
+## Scope
+
+Due to the variety of platforms out there, there is no single
+control plane implementation that can satisfy everyone's needs.  Hence this
```
Comment (htuch) [inline README.md]: Nit: double space `.  Hence`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,33 @@
...
+  server is responsible for pushing configuration updates to
+  Envoys. Consumers should be able to import this go library and use the
+  API server as is, in production deployments.
+
+* _Configuration Cache:_ The library will cache Envoy configurations in
+memory in an attempt to provide fast response to consumer Envoys. It is the
+responsibility of the consumer of this library to populate the cache as
+well as invalidate it when necessary. The cache will be keyed based on a
+pre-defined hash function whose keys are based on the
+[Node](https://github.com/envoyproxy/data-plane-api/blob/master/api/base.proto).
```
Comment (htuch) [inline README.md]: Nit: link to a specific line number and hash as best practice.

Comment (rshriram) [reply README.md]: This was intentional, as I didn't know if the line numbers would remain same, as we change the protos. Once we have documentation, we could link to the Node message's documentation. WDYT?

Comment (htuch) [reply README.md]: You can just link to a specific hash (GH even has some  click on line, click on ... on LHS, select "Copy Permalink"), e.g. https://github.com/envoyproxy/data-plane-api/blob/d4988844024d0bcff4bcd030552eabe3396203fa/api/base.proto#L26.

**Comment by tomwans** [comment]
LGTM

File: README.md (hunk 3 above)
Comment (9len) [reply README.md]: You can do a range of lines too:

https://github.com/envoyproxy/data-plane-api/blob/d4988844024d0bcff4bcd030552eabe3396203fa/api/base.proto#L26-L36

Comment (rshriram) [reply README.md]: Ah I didn't know the range of lines trick. Thats neat.. updated

Comment (9len) [reply README.md]: it's newish. we're pretty enamored with it.

**Comment by theatrus** [comment]
LGTM

**Comment by rosenhouse** [comment]
LGTM, excited to see this come together / possibly contribute where possible.

**Comment by mattklein123** [comment]
OK I'm going to go ahead and merge this. @rshriram @junr03 @kyessenov you all have commit access to this repo (along with existing @envoyproxy/maintainers). Enjoy! :)

**Pull request merged**